homeassistant/components/nws/sensor.py

```diff
@@ -0,0 +1,162 @@
+"""Sensors for National Weather Service (NWS)."""
+import logging
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_DEVICE_CLASS,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    LENGTH_KILOMETERS,
+    LENGTH_METERS,
+    LENGTH_MILES,
+    PRESSURE_INHG,
+    PRESSURE_PA,
+    SPEED_MILES_PER_HOUR,
+)
+from homeassistant.util.distance import convert as convert_distance
+from homeassistant.util.dt import utcnow
+from homeassistant.util.pressure import convert as convert_pressure
+
+from . import base_unique_id
+from .const import (
+    ATTR_ICON,
+    ATTR_LABEL,
+    ATTR_UNIT,
+    ATTR_UNIT_CONVERT,
+    ATTRIBUTION,
+    CONF_STATION,
+    COORDINATOR_OBSERVATION,
+    DOMAIN,
+    NWS_DATA,
+    OBSERVATION_VALID_TIME,
+    SENSOR_TYPES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the NWS weather platform."""
+    hass_data = hass.data[DOMAIN][entry.entry_id]
+    station = entry.data[CONF_STATION]
+    for sensor_name, sensor_data in SENSOR_TYPES.items():
+        if hass.config.units.is_metric:
+            unit = sensor_data[ATTR_UNIT]
+        else:
+            unit = sensor_data[ATTR_UNIT_CONVERT]
+        async_add_entities(
+            [
+                NWSSensor(
+                    entry.data,
+                    hass_data,
+                    sensor_name,
+                    station,
+                    sensor_data[ATTR_LABEL],
+                    sensor_data[ATTR_ICON],
+                    sensor_data[ATTR_DEVICE_CLASS],
+                    unit,
+                ),
+            ],
+            False,
+        )
+
+
+class NWSSensor(SensorEntity):
+    """An NWS Sensor Entity."""
+
+    def __init__(
+        self, entry_data, hass_data, name, station, label, icon, device_class, unit
+    ):
+        """Initialise the platform with a data instance."""
+        self._nws = hass_data[NWS_DATA]
+        self._latitude = entry_data[CONF_LATITUDE]
+        self._longitude = entry_data[CONF_LONGITUDE]
+        self._coordinator = hass_data[COORDINATOR_OBSERVATION]
+        self._name = name
```

Comment:
I'd rename this to `sensor_type`. Name is most often user configurable so it's easy to misunderstand it. That's not what this is.